oclint-rules/include/oclint/AbstractASTVisitorRule.h

```diff
@@ -12,7 +11,41 @@ template<typename T>
 class AbstractASTVisitorRule : public AbstractASTRuleBase, protected clang::RecursiveASTVisitor<T>
 {
     friend class clang::RecursiveASTVisitor<T>;
+private:
+    virtual bool shouldTraverseFile(clang::SourceLocation startLocation)
+    {
+        clang::SourceManager *sourceManager = &_carrier->getSourceManager();
+
+        if (sourceManager->getMainFileID() == sourceManager->getFileID(startLocation))
+        {
+            return true;
+        }
+
+        if (!visitHeaders())
+        {
+            return false;
+        }
+
+        llvm::StringRef sourceFileName = sourceManager->getFilename(startLocation);
+        return isAnalyzeFileHeadFile(sourceFileName);
+    }
+
+    virtual bool isAnalyzeFileHeadFile(llvm::StringRef fileNameStrRef) {
+        return true;
```

Comment:
This part is a bit confusing, can you revise it?

Reply:
I deleted the wrong code`return true;` and add some comments, hope it can be better now.